### DISCUSSION TITLE
Why are the mounts set to mode 02777?

### DISCUSSION THREAD

**Discussion by lel-amri** [topolvm/topolvm#981] [Q&A] [answered]
Hello,

Thanks for this project, I'm using it through the RedHat OpenShift's LVMS operator.

I'm wondering what is the rationale behind mounting the volumes as mode 02777. I could track the introduction of the setgid bit to https://github.com/topolvm/topolvm/commit/27623244501f39241683cb089150604535117478, but I'm lacking some context.

I understand that fsGroup cannot be used here, because two Pods with different fsGroup values running on the same node cannot coexist: a LV cannot be mounted more than once, which would be required to have different owners or modes.

But with supplementaryGroups available, I'd think that the volume could be mounted as 0777 and groups managed with the supplementaryGroups option. It seems to me that this way we're closer to how a user would expect a "classical" volume to work except for one thing: anyone that has access to the root of the mount can modify entries there. I think that the current situation doesn't play nice with supplementaryGroups and runAsGroup options.

Am I missing something?

**Comment by ushitora-anqou** [comment]
Hi. Thank you for your feedback. I think this is a historical thing. I'm investigating why this change was made, and I'll let you know if I find something.

**Comment by ushitora-anqou** [comment, marked as answer]
We use `02777` to improve the startup time of Pods using `fsGroup`.

If the files and directories in a volume don't have the GID specified in `fsGroup`, Kubernetes needs to recursively change their ownership during the Pod startup. This process can take a lot of time.

To solve this, we use the setgid bit (the '02' part of 02777). This ensures that any new files and directories created in the volume automatically have the same GID as `fsGroup`. As a result, Kubernetes doesn't need to perform the time-consuming chown operation during the Pod startup.
